### PR TITLE
Test extension for reproducing issue with query condition

### PR DESCRIPTION
**Pull request by jwillemsen**
Extended test with new dispose_qc commandline option that uses a query condition to read the data in the listener. This should be similar to the dispose test case where we get one dispose but in case of a query condition we don't get a dispose but a no data when we try to read the dispose data. Not fixed yet, this is just the test extension for the moment

### CODE REVIEW

**Comment by jwillemsen** [comment]
This is just the test extension, PR #590 has the same test extensions and a proposed fix

File: tests/DCPS/FooTest3_0/SubDriver.cpp [hunk 1]
```diff
@@ -173,6 +197,25 @@ SubDriver::init(int& argc, ACE_TCHAR* argv[])
...
+#ifndef OPENDDS_NO_QUERY_CONDITION
+  if (qc_usage_)
+  {
+    DDS::StringSeq params(1);
+    params.length (1);
+    params[0] = "101010";
+
+    DDS::QueryCondition_var qc =
+      datareader_->create_querycondition (DDS::NOT_READ_SAMPLE_STATE,
+                                          DDS::NEW_VIEW_STATE | DDS::NOT_NEW_VIEW_STATE,
```
Comment (mitza-oci) [inline tests/DCPS/FooTest3_0/SubDriver.cpp]: Isn't this just ANY_VIEW_STATE?

File: tests/DCPS/FooTest3_0/SubDriver.h [hunk 2]
```diff
@@ -12,6 +12,7 @@
 #include <vector>
 
 class DataReaderListenerImpl;
+class DataReaderQCListenerImpl;
```
Comment (mitza-oci) [inline tests/DCPS/FooTest3_0/SubDriver.h]: The forward declaration seems to be unused.

Comment (jwillemsen) [inline tests/DCPS/FooTest3_0/SubDriver.h]: Removed

File: tests/DCPS/FooTest3_0/SubDriver.cpp (hunk 1 above)
Comment (jwillemsen) [inline tests/DCPS/FooTest3_0/SubDriver.cpp]: Yes

**Comment by jwillemsen** [comment]
Closing, got this into #590

**Pull request closed without merging**